### ISSUE TITLE
[Bug] Importing the project in Unity 2019.4 gives errors

### ISSUE THREAD

**Issue by knightcube** [Unity-Technologies/arfoundation-samples#591]
**Unity bug report case number**
Please log a bug at https://unity3d.com/unity/qa/bug-reporting and post the case number here.

**Describe the bug**
I cloned the repository and tried opening it in Unity 2019.4. Once unity launches I get errors. I have attached the logs below

```
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\UserInterface\Bootstrap.cs(23,20): error CS0117: 'Collab' does not contain a definition for 'ShowChangesWindow'
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(108,22): error CS1061: 'Collab' does not contain a definition for 'ChangeItemsChanged' and no accessible extension method 'ChangeItemsChanged' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(109,22): error CS1061: 'Collab' does not contain a definition for 'SelectedChangeItemsChanged' and no accessible extension method 'SelectedChangeItemsChanged' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(110,22): error CS1061: 'Collab' does not contain a definition for 'RevisionUpdated_V2' and no accessible extension method 'RevisionUpdated_V2' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(112,22): error CS1061: 'Collab' does not contain a definition for 'CollabInfoChanged' and no accessible extension method 'CollabInfoChanged' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(114,22): error CS1061: 'Collab' does not contain a definition for 'ErrorOccurred_V2' and no accessible extension method 'ErrorOccurred_V2' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(460,36): error CS1061: 'Collab' does not contain a definition for 'GetChangesToPublish_V2' and no accessible extension method 'GetChangesToPublish_V2' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(477,22): error CS1061: 'Collab' does not contain a definition for 'PublishAssetsAsync' and no accessible extension method 'PublishAssetsAsync' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(503,32): error CS1061: 'RevisionsService' does not contain a definition for 'FetchSingleRevisionCallback' and no accessible extension method 'FetchSingleRevisionCallback' accepting a first argument of type 'RevisionsService' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(506,36): error CS1061: 'RevisionsService' does not contain a definition for 'GetRevision' and no accessible extension method 'GetRevision' accepting a first argument of type 'RevisionsService' could be found (are you missing a using directive or an assembly reference?)
Library\PackageCache\com.unity.textmeshpro@3.0.1\Scripts\Runtime\TMP_DefaultControls.cs(191,22): error CS1061: 'RectMask2D' does not contain a definition for 'padding' and no accessible extension method 'padding' accepting a first argument of type 'RectMask2D' could be found (are you missing a using directive or an assembly reference?)

```
**Expected behavior**
Expected the project to run without errors

**Smartphone (please complete the following information):**
 - Unity version - 2019.4
 - ARFoundation version - preview.7 - 4.1.0

**Comment by mzielke** [comment]
That is not a bug but a version mismatch indication for the Unity Collaborate Package. Go Window/package Manager/Unity Collaborate and set the Version from .7 - 4.1.0 to upgrade to 2019.x your current Unity Editor.

**Comment by RobG88** [comment]
I ran into the issue when down grading a project from Unity 2020 to 2019, as @mzielke said, in Unity, goto Windows => Package Manager, when the Package Manager is open, scroll down to Unity Collaborate, your current set version is incompatible with 2019, so use the drop down triangle and you should see additional options, choose the version which is 2019 Certified and in the bottom right make sure to click UP TO DATE.
![Unity Collaborate](https://user-images.githubusercontent.com/12281765/91750779-a9379980-eb91-11ea-8f75-7c3e28e2d5dd.PNG)

**Comment by knightcube** [comment]
Yes. I got it working now. Thanks a lot for your help :)

**Comment by harshitmac** [comment]
Faced the same problem but the fix does not apply to me the collaborate is uptodate

**Comment by mzielke** [comment]
What is the exact error message text you get ?

**Comment by harshitmac** [comment]
<img width="960" alt="Screenshot 2020-09-21 150040" src="https://user-images.githubusercontent.com/64486997/93751914-7f671500-fc1b-11ea-87be-3b12f31a898a.png">

**Comment by harshitmac** [comment]
1} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\UserInterface\Bootstrap.cs(23,20): error CS0117: 'Collab' does not contain a definition for 'ShowChangesWindow'

2} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(108,22): error CS1061: 'Collab' does not contain a definition for 'ChangeItemsChanged' and no accessible extension method 'ChangeItemsChanged' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

3} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(109,22): error CS1061: 'Collab' does not contain a definition for 'SelectedChangeItemsChanged' and no accessible extension method 'SelectedChangeItemsChanged' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

4} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(110,22): error CS1061: 'Collab' does not contain a definition for 'RevisionUpdated_V2' and no accessible extension method 'RevisionUpdated_V2' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

5} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(112,22): error CS1061: 'Collab' does not contain a definition for 'CollabInfoChanged' and no accessible extension method 'CollabInfoChanged' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

6} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(114,22): error CS1061: 'Collab' does not contain a definition for 'ErrorOccurred_V2' and no accessible extension method 'ErrorOccurred_V2' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

7} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(460,36): error CS1061: 'Collab' does not contain a definition for 'GetChangesToPublish_V2' and no accessible extension method 'GetChangesToPublish_V2' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?) 

8} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(477,22): error CS1061: 'Collab' does not contain a definition for 'PublishAssetsAsync' and no accessible extension method 'PublishAssetsAsync' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

9} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(503,32): error CS1061: 'RevisionsService' does not contain a definition for 'FetchSingleRevisionCallback' and no accessible extension method 'FetchSingleRevisionCallback' accepting a first argument of type 'RevisionsService' could be found (are you missing a using directive or an assembly reference?)

10} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(506,36): error CS1061: 'RevisionsService' does not contain a definition for 'GetRevision' and no accessible extension method 'GetRevision' accepting a first argument of type 'RevisionsService' could be found (are you missing a using directive or an assembly reference?)

11} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(511,36): error CS1061: 'RevisionsService' does not contain a definition for 'FetchSingleRevisionCallback' and no accessible extension method 'FetchSingleRevisionCallback' accepting a first argument of type 'RevisionsService' could be found (are you missing a using directive or an assembly reference?)

12} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(517,36): error CS1061: 'RevisionsService' does not contain a definition for 'FetchSingleRevisionCallback' and no accessible extension method 'FetchSingleRevisionCallback' accepting a first argument of type 'RevisionsService' could be found (are you missing a using directive or an assembly reference?)

13} Library\PackageCache\com.unity.collab-proxy@1.3.9\Editor\Models\Providers\Collab.cs(593,22): error CS1061: 'Collab' does not contain a definition for 'RevertFiles' and no accessible extension method 'RevertFiles' accepting a first argument of type 'Collab' could be found (are you missing a using directive or an assembly reference?)

14} Library\PackageCache\com.unity.textmeshpro@3.0.1\Scripts\Runtime\TMP_DefaultControls.cs(191,22): error CS1061: 'RectMask2D' does not contain a definition for 'padding' and no accessible extension method 'padding' accepting a first argument of type 'RectMask2D' could be found (are you missing a using directive or an assembly reference?)

**Comment by tdmowrer** [comment]
In general, you cannot downgrade a Unity project because the packages listed in a project's manifest are not necessarily compatible with an earlier version. So in this case, the collab and textmeshpro packages used by 2020.1 are not compatible with 2019.4.

If your goal is to open this repo (built for 2020.1) in 2019.4, you can try this:

1. Start a new Unity project in 2019.4.
2. Open the `Packages/manifest.json` file in a text editor.
3. For any package that is giving you trouble, (`com.unity.collab-proxy` & `com.unity.textmeshpro` in this case), copy the version from 2019.4's manifest into the ARFoundation project that you've tried to downgrade to 2019.4.

**Comment by harshitmac** [comment]
Thank you very much.

**Comment by paulmasri** [comment]
An alternative solution that might be a bit quicker is to simply remove and re-install packages **Unity Collaborate** and **TextMeshPro** from within the **Package Manager**. This worked for me as it installs the highest version _that is compatible with 2019.4_.

In my case, with a 2D project, I also had to do the same with **2D Animation**, **2D PSD Importer**, **2D SpriteShape** and **2D Common**.

**Comment by mansoortariq** [comment]
I have done changes in collab version , But still facing this error . Kindly any solution for this?

Library\PackageCache\com.unity.textmeshpro@3.0.1\Scripts\Editor\TMP_PackageUtilities.cs(453,84): error CS0103: The name 'VersionControlSettings' does not exist in the current context

**Comment by paulmasri** [comment]
@mansoortariq It sounds like you've done half of what has been suggested above — you need to downgrade both TextMeshPro and Unity Collaborate.

**Comment by Prashant29** [comment]
> I have done changes in collab version , But still facing this error . Kindly any solution for this?
> 
> Library\PackageCache\com.unity.textmeshpro@3.0.1\Scripts\Editor\TMP_PackageUtilities.cs(453,84): error CS0103: The name 'VersionControlSettings' does not exist in the current context

This is an Issue with the TextMeshPro plugin, please try to update to the verified version.

**Comment by URETY** [comment]
Yes. I got it working now. Thanks a lot for your help :)

**Comment by junaidshad** [comment]
i just removed and reinstalled that package and it worked for me just fine.

**Comment by q-viper** [comment]
For my case, I removed the unity collaborate and it worked.

**Comment by AhmedWagdyDevv** [comment]
I love github

**Comment by Nasir3** [comment]
I removed the unity collaborate and it worked.

**Comment by darelahmed** [comment]
> 
> 
> In general, you cannot downgrade a Unity project because the packages listed in a project's manifest are not necessarily compatible with an earlier version. So in this case, the collab and textmeshpro packages used by 2020.1 are not compatible with 2019.4.
> 
> If your goal is to open this repo (built for 2020.1) in 2019.4, you can try this:
> 
>     1. Start a new Unity project in 2019.4.
> 
>     2. Open the `Packages/manifest.json` file in a text editor.
> 
>     3. For any package that is giving you trouble, (`com.unity.collab-proxy` & `com.unity.textmeshpro` in this case), copy the version from 2019.4's manifest into the manifest of the ARFoundation project that you've tried to downgrade to 2019.4.

Thank you very much

**Comment by Keerthikkamath** [comment]
Downgrading the project from 2020.3 to 2019.4 threw errors wrt Collab-

**SOLUTION:**

**Windows -> Package Manager -> Advanced -> Reset Packages to defaults .**

This FIXED all my errors and now the project works fine.

**Comment by SquizzMe** [comment]
> I ran into the issue when down grading a project from Unity 2020 to 2019, as @mzielke said, in Unity, goto Window => Package Manager, when the Package Manager is open, scroll down to Unity Collaborate, your current set version is incompatible with 2019, so use the drop down triangle and you should see additional options, choose the version which is 2019 Certified and in the bottom right make sure to click UP TO DATE. ![Unity Collaborate](https://user-images.githubusercontent.com/12281765/91750779-a9379980-eb91-11ea-8f75-7c3e28e2d5dd.PNG)

for some reason my unity doesn't have that option of the unity collab. SquizzMe#384
![unity help](https://user-images.githubusercontent.com/95935502/145623793-98444319-7177-4d96-87cf-f5b5ef77407b.PNG)
8

**Comment by Charli3Trinh** [comment]
> > I ran into the issue when down grading a project from Unity 2020 to 2019, as @mzielke said, in Unity, goto Window => Package Manager, when the Package Manager is open, scroll down to Unity Collaborate, your current set version is incompatible with 2019, so use the drop down triangle and you should see additional options, choose the version which is 2019 Certified and in the bottom right make sure to click UP TO DATE. ![Unity Collaborate](https://user-images.githubusercontent.com/12281765/91750779-a9379980-eb91-11ea-8f75-7c3e28e2d5dd.PNG)
> 
> for some reason my unity doesn't have that option of the unity collab. SquizzMe#384 ![unity help](https://user-images.githubusercontent.com/95935502/145623793-98444319-7177-4d96-87cf-f5b5ef77407b.PNG) 8

Unity 2019.4 don't have that plugin:

Versions
Collaborate 1.3.0 fully supports Unity versions 2020.1 and higher.
Collaborate 1.2.16 fully supports Unity versions 2017.1 to 2019.3.
Collaborate is in beta for Unity versions lower than 2017.1.

**Comment by aldrick** [comment]
Weird I didn't see the Unity Collaborate in the package manager.
![image](https://user-images.githubusercontent.com/6886200/148082331-b1bca9c5-a8b5-4273-9b0a-32ed45433a0d.png)

**Comment by Sunwer91** [comment]
I'm in the same case, I don't have the Unity Collaborate in the Package Manager. :/

But I tried to **Reset packages to Default** and everything is fine now!
![image](https://user-images.githubusercontent.com/9313429/150912181-a4a3cead-be7c-45ab-8c16-70f75cbfeca5.png)

**Comment by Yarik1902** [comment]
HI, guys. I found that "Unity Collaborate" was renamed to "Version Control".

**Comment by Samuelincoom** [comment]
> Weird I didn't see the Unity Collaborate in the package manager. ![image](https://user-images.githubusercontent.com/6886200/148082331-b1bca9c5-a8b5-4273-9b0a-32ed45433a0d.png)

Unity collaborate manager was renamed to "Version control"

**Comment by ManChrys** [comment]
How i fix the error on my project (unity 2019.4.2f1) .

i update the textmeshpro

![text](https://user-images.githubusercontent.com/67521662/180400507-d2d88d7d-a21e-4b3e-b1b1-1b9f58c3cb86.jpg)

and the unity collaborate

![collab](https://user-images.githubusercontent.com/67521662/180400695-d86ed9ec-7750-479e-afa8-5dfa45caf06a.jpg)

and it fixed.
